code/modules/shuttle/shuttle.dm

```diff
@@ -297,6 +299,15 @@
 	initial_engines = count_engines()
 	current_engines = initial_engines
 
+	for (var/turf/T in all_turfs)
```

Comment:
you want to loop through all the shuttle areas, not all turfs here. Otherwise you include other shuttles if they're nearby